src/scatter.jl

```diff
@@ -0,0 +1,46 @@
+export scatter!
+
+const IntOrTuple = Union{Integer,Tuple}
+
+"""
+    scatter!(op, ys, us, xs)
+
+Scatter operation.
+
+    ys[xs[k]...] = (op).(ys[xs[k]...], us[k...])
+
+# Arguments
+- `op`: operations to be applied on `ys` and `us`, e.g. `+`, `-`, `*`, `/`, `max`, `min` and `mean`.
```

Comment:
since `mean` will be doing something different from the line of code, we should mention it separately